filesystem/src/main/java/org/frankframework/filesystem/FileSystemActor.java

```diff
@@ -381,10 +387,14 @@ public Message doAction(@Nonnull Message input, ParameterValueList pvl, @Nonnull
 					return result;
 				}
 				case LIST: {
-					String folder = arrangeFolder(determineInputFoldername(input, pvl));
-
-					ArrayBuilder directoryBuilder = DocumentBuilderFactory.startArrayDocument(getOutputFormat(), "directory", "file");
-					try (directoryBuilder; Stream<F> stream = FileSystemUtils.getFilteredStream(fileSystem, folder, getWildcard(), getExcludeWildcard())) {
+					String folder = arrangeFolder(determineInputFolderName(input, pvl));
+					determineTypeFilter(pvl);
+					String title = switch (typeFilter) {
+						case FILES_ONLY, FILES_AND_FOLDERS -> "file";
+						case FOLDERS_ONLY -> "folder";
+					};
+					ArrayBuilder directoryBuilder = DocumentBuilderFactory.startArrayDocument(getOutputFormat(), "directory", title);
+					try (directoryBuilder; Stream<F> stream = FileSystemUtils.getFilteredStream(fileSystem, folder, getWildcard(), getExcludeWildcard(), typeFilter)) {
```

Comment:
Just added the extra attribute `kind`, because the (S)FTP file systems have issues with `type`. The `isFolder()` is already added in previous commit.